wp-cache-sanitizer.php

```diff
@@ -0,0 +1,726 @@
+        case "cache_badbehaviour":
+            $value = $value == '' || $value == 'no' ? 0 : $value;
+            $sanitized_value = intval( $value );
+            break;
+
+        case "cache_awaitingmoderation":
+        case "cache_wptouch":
+        case "cache_domain_mapping":
+            $value = $value == '' ? '0' : $value;
+            $sanitized_value = intval( $value );
+            break;
+
+        case "cache_time_interval":
+            $sanitized_value = 600;
+            $value = is_numeric( $value ) ? intval( $value ) : $sanitized_value;
+            if ( $sanitized_value !== $value )
+                $sanitized_value = $value;
+            break;
+
+        case "cache_max_time":
+            $sanitized_value = 3600;
+            $value = is_numeric( $value ) ? intval( $value ) : $sanitized_value;
+            if ( $sanitized_value !== $value )
+                $sanitized_value = $value;
+            break;
+
+        case "cache_rejected_uri":
+            global $cache_rejected_uri;
+            foreach ( $value as $index => $url ) {
+                $value[ $index ] = str_replace( '\\\\', '\\', $url );
+            }
+            $sanitized_value = wp_cache_sanitize_value( implode(', ', $value), $cache_rejected_uri );
+            break;
+
+        case "wp_cache_preload_interval":
+            $sanitized_value = is_numeric( $value ) && $value >= 30 ? intval( $value ) : 0;
+            break;
+
+        case "sem_id":
+            global $WPSC_HTTP_HOST, $cache_path;
+            if ( preg_match('/^\d{9}$/', intval( $value ) ) ) {
+                $sanitized_value = intval( $value );
+            } else {
+                $cache_path =  '/' != substr( $cache_path, -1 )  ? $cache_path .= '/' : '';
+                $sanitized_value = crc32( $WPSC_HTTP_HOST . $cache_path ) & 0x7fffffff;
+            }
+            break;
+
+        default:
+            $sanitized_value = apply_filters( "wp_super_cache_sanitize_$value", $value );
+            $sanitized_value = intval( $sanitized_value );
+            break;
+    }
+    return $sanitized_value;
+  }
+
+  /**
+   * This sanitizes the settings where the allowed values are only strings.
+   * The sanitation method is different per setting name.
+   *
+   * @param String $setting The current string setting to be sanitized
+   * @param Mixed $value The mixed value of the current setting that will be sanitized
+   *
+   * @uses  wp_cache_sanitize_value Sanitizes an array value
+   *
+   * @return String|Array A sanitized version of the inputed value
+   */
+  function sanitize_string( $setting, $value ) {
+    switch ( $setting ) {
+
+        case 'cache_acceptable_files' :
+            global $cache_acceptable_files;
+            if ( is_array( $value ) ) {
+                $value = implode( ',', $value );
+            }
+            $sanitized_value = wp_cache_sanitize_value( $value, $cache_acceptable_files );
+            break;
+
+        case 'cache_badbehaviour_file':
```

Comment:
I did this with this setting and another one to explicitly show them being imported. Based on how the value is generated [here](https://github.com/Automattic/wp-super-cache/blob/master/plugins/badbehaviour.php#L57), I decided to import it but limit it to that value. This way someone can't edit the JSON file with a separate value for `cache_badbehaviour_file`.